rust-runtime/aws-smithy-http/src/result.rs

```diff
@@ -122,6 +122,14 @@ impl<E, R> ServiceError<E, R> {
     }
 }
 
+/// Constructs the unhandled variant of a code generated error.
+///
+/// This trait exists so that [`SdkError::into_service_error`] can be infallible.
+pub trait CreateUnhandledError {
+    /// Creates an unhandled error variant with the given `source`.
+    fn create_unhandled_error(source: Box<dyn Error + Send + Sync + 'static>) -> Self;
+}
```

Comment:
I wonder if this should be from `SdkError` to enable the future possibility of more precise handling?